### ISSUE TITLE
Regarding text -> image retrieval

### ISSUE THREAD

**Issue by xiankgx** [TIGER-AI-Lab/VLM2Vec#28]
Dear authors, great job on your work. I've been trying out the models for text to image retrieval task lately. Despite what's being reported on your paper beating OpenCLIP model for this task, I found it's performance to be weaker based on an internal image-text dataset. It's perhap on par or even worse than OpenAI CLIP. Perhaps including contrastive training as part of your multi task training would help in this regards?

**Comment by wenhuchen** [comment]
Thanks for the information. Have you tried the strongest llava checkpoint at https://huggingface.co/TIGER-Lab/VLM2Vec-LLaVa-Next?

Also, it's highly sensitive to the domain. Would you mind sharing some info in terms of the kind of the images you are dealing with? Like natural photographs, pdfs, documents, html or anything else?

**Comment by xiankgx** [comment]
Hi @wenhuchen , ya, I tried the two variants and I think Llava-Next version indeed performs better.

To give you context, I've sampled 10k image+caption pairs and performed the embedding representation.
Then I have a function that embeds the images as *target* representations using "<|image_1|> Represent the given image."
Then I have another function that embeds the captions as *query* representations using "Find me an everyday image that matches the given caption: {caption}".

For Llava-next, I replace <|image_1|> with <image>. Was earlier using the same and found that no error was repeated, and speed was very high and performance very poor. Hence, I think this is just an issue of not using the image data for representation.

The accuracy for them was 39-44% top-1 accuracy. For comparison with OpenCLIP and derived models, the performance was 70-80+ percent, depending on the model and pretrained data variants.

I then tried building a nearest neighbor search. Trying to just search for tiger, some other non-tiger images appear in top 5.

**Comment by XMHZZ2018** [comment]
@xiankgx 

Thank you for sharing the context! It seems that your image dataset is daily images, which is similar to MS COCO and Flickr30K. From our observations with MS COCO and Flickr30K, we found that our model performs better than the CLIP family. Would it be possible for you to share your dataset with us? I would be happy to take a look and see what's happening here.

**Comment by xiankgx** [comment]
@XMHZZ2018 

Hi, happy to share test set, along with model API server code and evaluation notebook. [link](https://drive.google.com/drive/folders/1EKhV9GVSoNQF2WowTW2OiHw20-kC7mue?usp=drive_link)

I hope I have not written the code wrongly.

**Comment by xiankgx** [comment]
Hi authors, looking at evaluation/eval_flickr.py,

it seems you are using the "qry" parameter in lines 68 and 78. While I was expecting to use "qry" for embedding query, and "tgt" for embedding candidates/targets. Would this be the issue?

**Comment by XMHZZ2018** [comment]
@xiankgx 
Thanks for sharing all these information!
For the "qry" and "tgt" thing, I don't think it would be the issue since it won't make any difference.

        qry_reps = self.encode_input(qry) if qry else None  # (bsz_per_device, dim)
        tgt_reps = self.encode_input(tgt) if tgt else None # (bsz_per_device, dim)

I will take a look at your code and data and get back to you soon.

**Comment by xiankgx** [comment]
Just for sharing, with this OpenCLIP,
`"ViT-B-32", pretrained="laion2b_s34b_b79k"`

the results are
```
 # top-1 acc: 0.6335024329640749
 # top-5 acc: 0.8417020395486076
```

**Comment by xiankgx** [comment]
> [@xiankgx](https://github.com/xiankgx)
> 
> Thank you for sharing the context! It seems that your image dataset is daily images, which is similar to MS COCO and Flickr30K. From our observations with MS COCO and Flickr30K, we found that our model performs better than the CLIP family. Would it be possible for you to share your dataset with us? I would be happy to take a look and see what's happening here.

Hi, have you managed to reproduced the results using my dataset?

I think perhaps the issue here is not using a a large scale text-image dataset like LAION for further finetuning.
Sometimes the performance is not related to the method (CLIP family, or VLM based embeddings) but more on the training dataset used and the scale of the training. For e.g., similarly models from CLIP vs OpenCLIP have huge difference in their retrieval performance.

**Comment by xiexie123** [comment]
any update?
